python/docs/source/usage.rst

```diff
@@ -476,8 +530,8 @@ to append pass in ``mode='append'``:
 
 .. code-block:: python
 
-    >>> write_deltalake('path/to/table', df, mode='overwrite')
-    >>> write_deltalake('path/to/table', df, mode='append')
+    >>> write_deltalake('path/to/table', df, mode='overwrite') # doctest: +SKIP
+    >>> write_deltalake('path/to/table', df, mode='append') # doctest: +SKIP
```

Comment:
Given the library is largely about reading and write from disk, it seems like it might be worth the effort to make this work. I was able to do this in lancedb/lance by configuring all doc tests to run in a temporary directory. This is just a few lines.

https://github.com/lancedb/lance/blob/84e4ce55f869a833c42239bfd6ba6176aa5840d3/python/python/lance/conftest.py#L18

Although the drawback is this might break the other tests that rely on reading specific pre-written tables in the repo. 🤔 I think that's a tradeoff I'd be willing to make. Let me know if you disagree.

Reply:
Great suggestion! It seems to work well in [commit](https://github.com/delta-io/delta-rs/pull/1840/commits/dc71553f210036ab6556a19066baae4cc8fed7e9) (on my machine, not sure about CI machine yet)